tests/unit/util/test_processor_stats.py

```diff
@@ -213,3 +214,48 @@ def test_log_to_prometheus_exports_calls_gauge_labels(self, mock_labels):
         mock_labels.assert_has_calls([mock.call(of="Dropper2"), mock.call().set(21)])
         mock_labels.assert_has_calls([mock.call(of="Dropper2"), mock.call().set(22)])
         mock_labels.assert_has_calls([mock.call(of="Dropper2"), mock.call().set(23)])
+
+    def test_add_per_processor_data_skips_excluded_processors(self):
+        self.status_tracker._pipeline[0].ps.aggr_data["processed"] = 13
+        self.status_tracker._pipeline[0].ps.aggr_data["warnings"] = 37
+        self.status_tracker._pipeline[1].ps.aggr_data["processed"] = 21
+        self.status_tracker._pipeline[1].ps.aggr_data["warnings"] = 21
+
+        clusterer = Clusterer(
+            "Clusterer1",
+            logger=self.logger,
+            tree_config="",
+            specific_rules="",
+            generic_rules="",
+            output_field_name="",
+        )
+        clusterer.ps.aggr_data["processed"] = 78
+
+        self.status_tracker._pipeline.append(clusterer)
+
+        process_data = {}
+        self.status_tracker._add_per_processor_data(process_data=process_data)
+
+        assert process_data.get("Dropper1", {}).get("matches") == 0
+        assert process_data.get("Dropper1", {}).get("processed") == 13
```

Comment:
Consider using the initialized value from dropper1

Reply:
done